src/index.ts

```diff
@@ -454,6 +454,11 @@ const cleanUpPopOut = (checkout: DinteroCheckoutInstance) => {
     }
 };
 
+const composeUrl = (base:string, path:string, query: string) => {
+    const slash = base.endsWith('/')? '': '/';
+    return `${base}${slash}${path}?${query}`;
+}
+
```

Comment:
```suggestion
const composeUrl = (base:string, path:string, query: Record<string, string>) => {
    const url = new URL(path, base);
    Object.entries(query).forEach(([name, value]) =>
        url.searchParams.append(name, value)
    );
    return url.href;
}
```
is there a reason to not use URL?

Reply:
Yes, the base here could also have a path we want to add to rather than replace.